classes/class.ilObjShortLink.php

```diff
@@ -0,0 +1,303 @@
+ * @version $Id$
+ */
+class ilObjShortLink {
+
+    /**
+     * @var ilDB $db
+     */
+    protected $db;
+    /**
+     * @var ilObjUser $usr
+     */
+    protected $usr;
+    /**
+     * @var ilShortLinkPlugin $pl
+     */
+    protected $pl;
+    /**
+     * @var int $id
+     */
+    protected $id = 0;
+    /**
+     * @var string $contact
+     */
+    protected $contact;
+    /**
+     * @var string $longURL
+     */
+    protected $longURL;
+    /**
+     * @var string $shortLink
+     */
+    protected $shortLink;
+    /**
+     * @var $customer
+     */
+    protected $customer;
+    /**
+     * @var $currentUserId;
+     */
+    protected $currentUserId;
+
+
+    public function __construct() {
+        global $ilDB, $ilUser;
+
+        $this->db = $ilDB;
+        $this->usr = $ilUser;
+
+        $this->pl = new ilShortLinkPlugin();
+
+        $this->currentUserId = $this->usr->getId();
+    }
+
+    /**
+     * Inserts new item into DB
+     */
+    public function doCreate() {
+        $stmt = $this->db->prepare('INSERT INTO ' . ilShortLinkPlugin::TABLE_NAME .
+            ' (id, short_link, full_url, customer, contact_user_login) VALUES (?, ?, ?, ?, ?);',
+            array('integer', 'text', 'text', 'text', 'text'));
+        $this->db->execute($stmt, array($this->getId(), $this->getShortLink(), $this->getLongURL(), $this->getCustomer(), $this->getContact()));
+    }
+
+    /**
+     * @param bool $as_obj
+     * @param int $id
+     * @return array $singleEntry
+     */
+    public function readSingleEntry($as_obj = TRUE, $id) {
+        $currentUser = $this->usr->getLogin();
+        $set = $this->db->query('SELECT * FROM ' . ilShortLinkPlugin::TABLE_NAME . ' WHERE id=' . $id);
+
+        $singleEntry = array();
+
+        while ($rec = $this->db->fetchAssoc($set)) {
+            if ($as_obj) {
+                if($currentUser == $rec['contact_user_login'] || $this->checkAdministrationPrivilegesFromDB()){
+                    $singleEntry[] = array('id'=>$rec['id'], 'long_url'=>$rec['full_url'], 'short_link'=>$rec['short_link'],
+                        'customer'=>$rec['customer'], 'contact'=>$rec['contact_user_login']);
+                }
+            } else {
+                $shortLinks[] = $rec;
+            }
+        }
+        return $singleEntry;
+    }
+
+    public function nextId() {
+        $this->setId($this->db->nextId(ilShortLinkPlugin::TABLE_NAME));
+    }
+
+    /**
+     * Get data from DB and put it into an array
+     *
+     * @param bool $as_obj
+     * @return array $shortLinks
+     */
+    public function readTablesPerUser($as_obj = TRUE) { // << ?? $as_obj reason?? >>
+        $shortLinks = array();
+        $currentUser = $this->usr->getLogin();
+
+        $set = $this->db->query('SELECT * FROM ' . ilShortLinkPlugin::TABLE_NAME);
+        $isAdministrator = $this->checkAdministrationPrivilegesFromDB();
+        while ($rec = $this->db->fetchAssoc($set)) {
```

Comment:
The only alternative I see right now is:     
$rec = $this->db->fetchAssoc($set);
        if (!empty($rec)) {}

But the official documentation of php and mysql show it as standard.

http://php.net/manual/de/mysqli-result.fetch-assoc.php
https://dev.mysql.com/doc/apis-php/en/apis-php-mysqli-result.fetch-assoc.html